### PR TITLE
Sample revision

### PR DESCRIPTION
**Pull request by derisen**
This is a function web api written in node.js and express framework. The client to call this web api is stored [here](https://github.com/Azure-Samples/ms-identity-javascript-callapi/pull/1)

This work falls under chapter 3 - authorization II of the chapter-wise tutorial

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,211 +1,239 @@
...
 products:
-- nodejs
-- azure
-- azure-active-directory
-name: NodeJS Azure Function Web API secured by Azure AD
-urlFragment: "ms-identity-nodejs-webapi-azurefunctions"
+  - nodejs
+  - azure-functions
+  - azure-active-directory
+name: A NodeJS Azure Function web API secured by Azure AD
```
Comment (kalyankrishna1) [inline README.md]: >web API [](start = 30, length = 7)

Web API (search and replace)

File: README.md [hunk 2]
```diff
@@ -1,211 +1,239 @@
...
+ 1. [More information](#more-information)
+ 1. [Community Help and Support](#community-help-and-support)
+ 1. [Contributing](#contributing)
+ 1. [Code of Conduct](#code-of-conduct)
 
-This code example demonstrates how to secure an Azure function with Azure AD when the function uses HTTPTrigger and exposes a Web API. The Web API is written using expressjs, and the authentication is provided by passport-azure-ad.
+## Overview
 
-This readme walks you through the steps of setting this code up in your Azure subscription.
+This sample demonstrates how to secure an [Azure Function]() with [Azure Active Directory (Azure AD)]() when the function uses an [HTTPTrigger]() and exposes a web API. The web API is written using the [express.js]() framework, and the authentication is provided by [passport-azure-ad]() library.
```
Comment (kalyankrishna1) [inline README.md]: >[Azure Function]() [](start = 42, length = 18)

Link?

Comment (kalyankrishna1) [inline README.md]: >[Azure Active Directory (Azure AD)]()  [](start = 66, length = 38)

Link?

Comment (kalyankrishna1) [inline README.md]: >written [](start = 184, length = 7)

uses the

File: README.md [hunk 3]
```diff
@@ -1,211 +1,239 @@
...
 
-To keep things simple, we will reuse the same app registration as both the client, and the API. This eliminates any need to provide explicit consent. For our code example here, the client will use [auth code flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow), for which we will also need a client secret. We are trying to mimic a web app calling this API, and a web app can act as a confidential client.
+> :warning: Given that the name of the sample is quite long, and so are the names of the referenced packages, you might want to clone it in a folder close to the root of your hard drive, to avoid maximum file path length limitations on Windows.
 
-To setup the app, you can use the below azure CLI script. Note the placeholders demarcated in `<..>` brackets. Make sure to replace them with your environment specific values.
+### Step 2: Install project dependencies
 
-``` SHELL
-az ad app create --display-name "FuncAPI" --credential-description "funcapi" --password "p@ssword1" --reply-urls "http://localhost:7071" --identifier-uris "https://funcapi.<tenantname>.onmicrosoft.com"
+```console
```
Comment (kalyankrishna1) [inline README.md:81]: >console [](start = 3, length = 7)

add the line
npm install -g Azure-functions-core-tools

File: Function/auth.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "clientID": "e767d418-b80b-4568-9754-557f40697fc5",
+    "tenantID": "cbaf2168-de14-4c72-9d88-f5f05366dbef",
+    "audience": "e767d418-b80b-4568-9754-557f40697fc5",
+    "scope": ["access_as_user"],
```
Comment (kalyankrishna1) [inline Function/auth.json:8]: >"scope": ["access_as_user"], [](start = 4, length = 28)

should not be overwritten

File: AppCreationScripts/sample.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            "value": "$tenantId"
+          },
+          {
+            "key": "audience",
+            "value": ".AppId"
+          },
+          {
+            "key": "scope",
+            "value": "service.Scope"
+          }
```
Comment (kalyankrishna1) [inline AppCreationScripts/sample.json:98]: remove this

File: README.md [hunk 6]
```diff
@@ -1,211 +1,239 @@
...
+1. The function app will run on `http://localhost:7071` when you test it locally.
+2. The function app will run on `https://<yournodejsfunction>.azurewebsites.net` when you run it deployed to Azure.
+
+You will need a **client** for calling the web API. Refer to the sample: [JavaScript single-page application calling a custom web API with MSAL.js 2.x using the auth code flow with PKCE](https://github.com/Azure-Samples/ms-identity-javascript-callapi).
+
+Once you install the **client** app, do:
+
+1. Open the `App\apiConfig.js` file.
+1. Find the key `Enter_the_Web_Api_Uri_Here` and replace the existing value with the coordinates of your web API (e.g. `http://localhost:7071/api`).
+1. Find the key `Enter_the_Web_Api_Scope_Here` and replace the existing value with the scopes for your web API (e.g. `api://e767d418-b80b-4568-9754-557f40697fc5/access_as_user`).
```
Comment (kalyankrishna1) [inline README.md]: >api://e767d418-b80b-4568-9754-557f40697fc5/access_as_user [](start = 118, length = 57)

Check this too

File: README.md [hunk 7]
```diff
@@ -1,211 +1,239 @@
...
+> - Describe where the code uses auth libraries, or calls the graph
+> - Describe specific aspects (e.g. caching, validation etc.)
+
+## Deployment
+
+### Deployment to Azure Functions
+
+There is one web project in this sample. To deploy it to **Azure Functions**, you'll need to:
+
+- create an **Azure Function**
```
Comment (kalyankrishna1) [inline README.md]: >an **Azure Function** [](start = 9, length = 21)

a Function App. Choose Node.js as "Runtime Stack".
---- more--

After deployment, make sure that "App Service Authentication" is switched off as we will Authorize the callers in the code itself.

File: Function/auth.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
+{
+    "clientID": "e767d418-b80b-4568-9754-557f40697fc5",
```
Comment (kalyankrishna1) [inline Function/auth.json]: >clientID [](start = 5, length = 8)

Authority

File: README.md [hunk 9]
```diff
@@ -1,211 +1,239 @@
...
+- publish the projects to the **Azure Functions**, and
+- update its client(s) to call the deployed function instead of the local environment.
```
Comment (kalyankrishna1) [inline README.md:329]: >environment [](start = 74, length = 11)

add these steps again

1. Open the `App\apiConfig.js` file.
1. Find the key `Enter_the_Web_Api_Uri_Here` and replace the existing value with the coordinates of your web API (e.g. `https://<yournodejsfunction>.azurewebsites.net`).

File: Function/MyHttpTrigger/index.js [hunk 10]
```diff
@@ -1,43 +1,57 @@
...
 
-var options = {
-    identityMetadata: "https://login.microsoftonline.com/" + tenantID + "/v2.0/.well-known/openid-configuration",
-    clientID: clientID,
-    issuer: "https://sts.windows.net/" + tenantID + "/",
-    audience: appIdURI,
-    loggingLevel: "info",
-    passReqToCallback: false
+const options = {
+    identityMetadata: `https://login.microsoftonline.com/${auth.tenantID}/v2.0/.well-known/openid-configuration`,
```
Comment (kalyankrishna1) [inline Function/MyHttpTrigger/index.js]: >https://login.microsoftonline.com [](start = 23, length = 33)

This should be from config

File: README.md (hunk 9 above)
Comment (derisen) [reply README.md:329]: this applies to the client SPA, which will be separate

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

**Review by Shama-K — APPROVED** [review]
LGTM, thanks.

**Pull request merged**